jerry-ext/handler/handler-assert.c

```diff
@@ -132,3 +160,213 @@ jerryx_handler_assert (const jerry_value_t func_obj_val, /**< function object */
+} /* jerryx_handler_assert_equal */
+
+
+/**
+ * @return true - if the two arguments are strict equal.
+ * If they aren't, this function calls jerryx_fatal.
+ *         Note that the function does not return otherwise.
+ */
+static jerry_value_t
+jerryx_handler_assert_strict_equal (const jerry_value_t func_obj_val, /**< function object */
+                                    const jerry_value_t this_p, /**< this arg */
+                                    const jerry_value_t args_p[], /**< function arguments */
+                                    const jerry_length_t args_cnt) /**< number of function arguments */
+{
+  return jerryx_handler_assert_equal_helper (func_obj_val, this_p, args_p, args_cnt, JERRYX_OP_STRICT_EQUAL);
+} /* jerryx_handler_assert_strict_equal */
+
+/**
+ * @return true - if the two arguments are not equal.
+ * If they are, this function calls jerryx_fatal.
+ *         Note that the function does not return otherwise.
+ */
+static jerry_value_t
+jerryx_handler_assert_not_equal (const jerry_value_t func_obj_val, /**< function object */
+                                 const jerry_value_t this_p, /**< this arg */
+                                 const jerry_value_t args_p[], /**< function arguments */
+                                 const jerry_length_t args_cnt) /**< number of function arguments */
+{
+  return jerryx_handler_assert_equal_helper (func_obj_val, this_p, args_p, args_cnt, JERRYX_OP_NOT_EQUAL);
+} /* jerryx_handler_assert_not_equal */
+
+/**
+ * return true - if the two arguments are not strict equal.
+ * If they are, this function calls jerryx_fatal.
+ *         Note that the function does not return otherwise.
+ */
+static jerry_value_t
+jerryx_handler_assert_not_strict_equal (const jerry_value_t func_obj_val, /**< function object */
+                                        const jerry_value_t this_p, /**< this arg */
+                                        const jerry_value_t args_p[], /**< function arguments */
+                                        const jerry_length_t args_cnt) /**< number of function arguments */
+{
+  return jerryx_handler_assert_equal_helper (func_obj_val, this_p, args_p, args_cnt, JERRYX_OP_NOT_STRICT_EQUAL);
+} /* jerryx_handler_assert_not_strict_equal */
+
+/**
+ * return message-if this function has 1 argument and the argument type is string.
+ * if this function hasn't got an argument call jerryx_fatal.
+ */
+static jerry_value_t
+jerryx_handler_assert_fail (const jerry_value_t func_obj_val, /**< function object */
+                            const jerry_value_t this_p, /**< this arg */
+                            const jerry_value_t args_p[], /**< function arguments */
+                            const jerry_length_t args_cnt) /**< number of function arguments */
+{
+  (void) func_obj_val; /* unused */
+  (void) this_p; /* unused */
+
+  if (jerry_value_is_string (args_p[0]) && args_cnt == 1)
```

Comment:
The number of arguments should be checked before accessing the arguments array, also I'd change the relation to `args_cnt > 0`